File: БР1.1/Захматов Юрий/labs/lab1/src/main/java/com/petproject/itmoacsbackend/users/dto/UserUpdateRequest.java
package com.petproject.itmoacsbackend.users.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;

public record UserUpdateRequest (

    String firstName,

    String lastName,

    String patronymic,

    @Pattern(
            regexp = "^\\+?[0-9. ()-]{7,25}$",
            message = "Phone number is invalid"
    )
    String phoneNumber,

    @Email
    String email
    )
{ }

File: БР1.2/Алексеев Андрей/labs/lab1/src/main/kotlin/ru/itmo/pxdkxvan/lab1/user/dto/UserUpdateRequest.kt
package ru.itmo.pxdkxvan.lab1.user.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Size

data class UserUpdateRequest(
    @field:Size(max = 100)
    val firstName: String? = null,
    @field:Size(max = 100)
    val lastName: String? = null,
    @field:Size(max = 100)
    val middleName: String? = null,
    @field:Email
    @field:Size(max = 255)
    val email: String? = null,
    @field:Size(max = 32)
    val phone: String? = null,
)
